src/components/KModal/KModal.vue

```diff
@@ -100,343 +87,270 @@
+
+defineOptions({
+  inheritAttrs: false,
+})
+
+const { getSizeFromString } = useUtilities()
 
 const props = defineProps({
-  /**
-  * Set the text of the title, if using title slot, this text is for the aria-label
-  */
-  title: {
-    type: String,
-    required: true,
-  },
-  /**
-  * Title is required for aria-labelling, toggle if the title is visible on the modal
-  */
-  hideTitle: {
+  visible: {
     type: Boolean,
     default: false,
   },
-  /**
-  * The dismiss icon is visible by default when using the `header-image` slot.
-  * Set to true to hide the 'x' dismiss button
-  */
-  hideDismissIcon: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-  * Controls whether the dismiss button is light or dark shade.
-  */
-  dismissButtonTheme: {
-    type: String as PropType<DismissButtonTheme>,
-    default: 'dark',
-    validator: (val: DismissButtonTheme): boolean => DismissButtonThemeArray.includes(val),
-  },
-  /**
-  * Set the text of the body content
-  */
-  content: {
+  title: {
     type: String,
     default: '',
   },
-  /**
-  * Set the alignment for the title and content
-  */
-  textAlign: {
-    type: String as PropType<TextAlign>,
-    default: 'center',
-    validator: (val: TextAlign): boolean => TextAlignArray.includes(val),
-  },
-  /**
-  * Pass whether or not the modal should be visible
-  */
-  isVisible: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-  * Set the text of the action/proceed button
-  */
   actionButtonText: {
     type: String,
     default: 'Submit',
   },
-  /**
-  * Set the appearance of the action/proceed button
-  */
   actionButtonAppearance: {
     type: String as PropType<ButtonAppearance>,
     default: 'primary',
   },
-  /**
-  * Set the text of the close/cancel button
-  */
+  actionButtonDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  hideCancelButton: {
+    type: Boolean,
+    default: false,
+  },
   cancelButtonText: {
     type: String,
     default: 'Cancel',
   },
-  /**
-  * Set the appearance of the close/cancel button
-  */
   cancelButtonAppearance: {
     type: String as PropType<ButtonAppearance>,
-    default: 'secondary',
+    default: 'tertiary',
   },
-  /**
-  * Set to not render the cancel button
-  */
-  hideCancelButton: {
+  cancelButtonDisabled: {
     type: Boolean,
     default: false,
   },
-  /**
-  * Options to be passed to tabbable
-  */
   tabbableOptions: {
     type: Object,
     default: () => ({}),
   },
-  /**
-  * Test mode - for testing only, strips out generated ids
-  */
-  testMode: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-   * The max-width of the modal. **Default: `500px`**.
-   */
-  maxWidth: {
+  width: {
     type: String,
     required: false,
     default: '500px',
   },
+  maxHeight: {
+    type: String,
+    required: false,
+    default: 'calc(100vh - 200px)',
+  },
+  closeOnBackdropClick: {
+    type: Boolean,
+    default: false,
+  },
+  proceedOnEnter: {
```

Comment:
There is some sort of bug here. In the sandbox, if I open the modal, don't focus it, and press `Enter` on my keyboard, the `proceed` event is emitted but the modal doesn't actually close.
![image](https://github.com/Kong/kongponents/assets/2229946/dcff60a1-e7c8-4a2c-97fb-11a10af0ae20)

Reply:
(I added a console log in the `proceed()` method to illustrate)